analyses/independent-samples/01-generate-independent-specimens.R

```diff
@@ -0,0 +1,108 @@
+# 01-generate-independent-specimens.R
+#
+# Josh Shapiro for CCDL 2019
+# 
+# Purpose: Generate tables of independent specimens where no two specimens are 
+#   chosen from the same individual.
+#
+# Option descriptions
+# -f, --histology_file : File path to where you would like the annotation_rds file to be
+#               stored
+# -o,--output_directory : Output directory
+# 
+# example invocation:
+# Rscript analyses/independent-samples/01-generate-independent-specimens.R \
+#   -f data/pbta-histologies.tsv \
+#   -o analyses/independent-samples/results
+
+
+# Base directories
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+analysis_dir <- file.path(root_dir, "analyses", "independent-samples")
+
+# Load the optparse library
+library(optparse)
+
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+# source sample selection function
+source(file.path(analysis_dir, "independent-samples.R"))
+
+set.seed(201910)
+
+# Parse options
+
+option_list <- list(
+  make_option(
+    c("-f", "--histology_file"),
+    type = "character",
+    default = NULL,
+    help = "path to the histology tsv file, relative to project root",
+  ),
+  make_option(
+    c("-o", "--output_directory"),
+    type = "character",
+    default = NULL,
+    help = "path to output directory, relative to project root"
+  )
+)
+
+opts <- parse_args(OptionParser(option_list = option_list))
+
+# set output files
+out_dir <- file.path(root_dir, opts$output_directory)
+if (!dir.exists(out_dir)){
+  dir.create(out_dir, recursive = TRUE)
+}
+
+wgs_primary_file <- file.path(out_dir, 
+                              "independent-specimens.wgs.primary.tsv")
+wgs_primplus_file <- file.path(out_dir, 
+                               "independent-specimens.wgs.primary-plus.tsv")
+wgswxs_primary_file <- file.path(out_dir, 
+                                 "independent-specimens.wgswxs.primary.tsv")
+wgswxs_primplus_file <- file.path(out_dir, 
+                                  "independent-specimens.wgswxs.primary-plus.tsv")
+
+# Read histology file
+sample_df <- readr::read_tsv(file.path(root_dir, opts$histology_file), 
+                             col_types = readr::cols()) # suppress parse message
+
+
+# Filter to only samples from tumors, where composition is known to be Solid Tissue
+# Note that there are some samples with unknown composition, but these will be ignored for now.
+tumor_samples <- sample_df %>%
+  dplyr::filter(sample_type == "Tumor", 
+                composition == "Solid Tissue", 
+                experimental_strategy %in% c("WGS", "WXS"))
+
+# Generate WGS independent samples
+wgs_samples <- tumor_samples %>%
+  dplyr::filter(experimental_strategy == "WGS")
+
+wgs_primary <- independent_samples(wgs_samples, tumor_types = "primary")
+wgs_primary_plus <- independent_samples(wgs_samples, tumor_types = "prefer_primary")
+
+# Generate lists for WXS only samples
+wxs_only_samples <-  tumor_samples %>% 
+  dplyr::filter(!(Kids_First_Participant_ID %in% 
```

Comment:
I would prefer if we could retain the completed set of WXS only samples rather than the limited WGS for these patients, since we are looking at coding regions for the majority of SNV analyses and so that we do not lose a large chunk of DIPG samples - one of the currently incurable tumors. For CNV/SV analyses, we have only called those on WGS, so it makes sense to prefer WGS there.

Reply:
Or, do you just want them as file lists of PT and BS ids, which would be pretty easy to add?

Reply:
There are 4 files, two of which have WGS + WXS for the samples where WGS is not available, so that use case should be covered. Do you think there is a reason to prefer WXS? Do we want a WXS-preferred list too?

Reply:
OK, that works! I think typically, I would say we could use one or the other as a validation cohort, but that is probably beyond the scope currently. Was thinking WXS preferred due to higher coverage and more TP calls, however, we do at least know the drivers are consistently called in WXS/WGS for that cohort. We can probably leave as is for now?